pkg/reconciler/deployment/deployment.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package deployment
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/knative-gcp/pkg/reconciler"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	"strconv"
+
+	v1 "k8s.io/api/apps/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+)
+
+const (
+	updatedSecretRevisionAnno = "api.v1.namespaces.cloud-run-events.secrets.google-cloud-key/updated-revision"
```

Comment:
Is this some convention we use? The order seems odd.

Reply:
I'm not sure whether we have any convention for the annotation. The first part before `/`, I just copied the self link of secret which is `/api/v1/namespaces/cloud-run-events/secrets/google-cloud-key`. Do you mean the order of the self link is odd?

Reply:
I see. It looks weird because the self link uses slashes in which case the bigger scope usually comes first (e.g. /google/cloud/knative). When you change the separator to dot, the convention usually is reversed - bigger scope comes after (e.g. knative.cloud.google). I'd suggest something like `google-cloud-key.cloud-run-events` but it's up to you.

Reply:
I think we can use `events.cloud.google.com/secret-update` or something like that

Reply:
and maybe call it `SecretUpdateAnnotation`